datastore.go

```diff
@@ -79,15 +79,25 @@ var DefaultOptions Options
 
 func init() {
 	DefaultOptions = Options{
-		GcDiscardRatio: 0.2,
+		GcDiscardRatio: 0.01,
 		GcInterval:     15 * time.Minute,
 		GcSleep:        10 * time.Second,
 		Options:        badger.LSMOnlyOptions(""),
 	}
-	// This is to optimize the database on close so it can be opened
-	// read-only and efficiently queried. We don't do that and hanging on
-	// stop isn't nice.
-	DefaultOptions.Options.CompactL0OnClose = false
+	// This is to optimize the database on closure
+	DefaultOptions.Options.CompactL0OnClose = true
+	
+	// Remove elements which the has been deleted from the database
+	DefaultOptions.Options.NumVersionsToKeep = 0
+	
+	// Reduce the number of zero tables (which are hold in memory)
+	DefaultOptions.Options.NumLevelZeroTables = 1
+	
+	// Reduce the number of zero tables which are stalled
+	DefaultOptions.Options.NumLevelZeroTablesStall = 2
```

Comment:
This will hurt write throughput. All this does is block writes until compaction happens. Given that compaction will happen eventually anyways, this setting won't help us.